flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/ScalingMetricEvaluator.java

```diff
@@ -137,12 +138,13 @@ private Map<ScalingMetric, EvaluatedScalingMetric> evaluateMetrics(
                         latestVertexMetrics.get(TRUE_PROCESSING_RATE),
                         getAverage(TRUE_PROCESSING_RATE, vertex, metricsHistory)));
 
-        evaluatedMetrics.put(
-                PARALLELISM, EvaluatedScalingMetric.of(topology.getParallelisms().get(vertex)));
+        var parallelism = EvaluatedScalingMetric.of(topology.getParallelisms().get(vertex));
+        evaluatedMetrics.put(PARALLELISM, parallelism);
+        evaluatedMetrics.put(RECOMMENDED_PARALLELISM, parallelism);
```

Comment:
I feel like we can't add it here yet because we don't know yet what the recommended parallelism is.

Reply:
We could simply set this in the scaling executor only (but initialize to null for the metric registration)

Reply:
Updated the requested changes